hack/update-mocks.sh

```diff
@@ -42,17 +42,22 @@ function git_grep() {
 
 cd "${KUBE_ROOT}"
 
-GENERATED_MOCK_FILE_REGEX="^// Code generated by MockGen. DO NOT EDIT.$"
+GENERATED_MOCK_FILE_REGEX="^// Code generated by mockery v[0-9.]\+. DO NOT EDIT.$"
 
 # pick a tempfile path for writing to
 tmp=$(mktemp)
 kube::util::trap_add "rm -f ${tmp:?}" EXIT
 
+# test/utils/oidc/testserver.go needs the mocks to build; mockery processes full packages,
+# and fails if testserver.go can't be built. Rename it out of the way for mock generation.
+kube::util::trap_add "mv test/utils/oidc/testserver.go.disabled test/utils/oidc/testserver.go" EXIT
```

Comment:
hrmm... @enj @aramase is there a way to tweak this test so this juggling isn't needed?

Reply:
Moving `testserver.go` or the generated mocks to another package would fix this.

Reply:
can we move the generated mocks to test/utils/oidc/mock/... as part of this PR? we're already regenerating completely new files, right?

Reply:
Yes, we are indeed. The mocks are only used in `testserver.go` so the change wouldn’t affect anyone else. While I’m at it I’ll fix the ugly `mock_jw_ks_handler.go` file name...

Reply:
this seems better, thanks